indico/core/db/sqlalchemy/protection.py

```diff
@@ -639,6 +649,31 @@ def _format_permissions(permissions):
                 data['Permissions (removed)'] = _format_permissions(removed_permissions)
             if current_permissions:
                 data['Permissions'] = _format_permissions(current_permissions)
-            obj.log(log_realm, LogKind.change, 'Protection', 'ACL entry changed', user, data=data)
+        else:
+            return
+
+        if log_realm:
+            obj.log(log_realm, log_kind, 'Protection', summary, user, data=data)
+
+        if principal.principal_type == PrincipalType.user:
+            user_log_data = data.copy()
+            del user_log_data['User']
+            obj_title = getattr(obj, 'title', None) or getattr(obj, 'name', str(obj))
+            if parent := (getattr(obj, 'category', None)
+                          or getattr(obj, 'event', None)
+                          or getattr(obj, 'location', None)):
```

Comment:
I think all of these objects may have a `protection_parent` property you can use instead :)